### PR TITLE
Add accuracy tests for RDT 

### PR DESCRIPTION
**Pull request by mihofer**
Similar to CRDT test, uses some provided lin files to do RDT processing and then check results against optics functions or PTC_RDT data.
Note: due to the standard processing of RDT with the next 3 BPMs being used for the reconstruction, only selected BPMs are used for the comparison

### CODE REVIEW

**Comment by mihofer** [comment]
Odd enough, the skew quadrupole B2 test fails as the imaginary part of the F1001 terms is in antiphase to the optics functions result.. not clear why

model tfs and processing script are attached (in some ungodly format, since GH doesnt like the .py extension)
[model.txt](https://github.com/pylhc/omc3/files/7144780/model.txt)
[create_model_coupling_b2.txt](https://github.com/pylhc/omc3/files/7144785/create_model_coupling_b2.txt)

**Comment by mihofer** [comment]
When and why was the special phase functionality "removed" https://github.com/pylhc/omc3/blob/8231cb57d70f3ad045820cda900c7ae17e2b1c58/omc3/optics_measurements/measure_optics.py#L52 ?

**Comment by fsoubelet** [comment]
> When and why was the special phase functionality "removed"/temporarily disabled?
> 
> https://github.com/pylhc/omc3/blob/8231cb57d70f3ad045820cda900c7ae17e2b1c58/omc3/optics_measurements/measure_optics.py#L52

`git blame` says @awegsche on 2021-07-13, maybe that was the change.

**Comment by awegsche** [comment]
> > When and why was the special phase functionality "removed"/temporarily disabled?
> > https://github.com/pylhc/omc3/blob/8231cb57d70f3ad045820cda900c7ae17e2b1c58/omc3/optics_measurements/measure_optics.py#L52
> 
> `git blame` says @awegsche on 2021-07-13, maybe that was the change.

oopsie, might have commented out this once

File: omc3/optics_measurements/phase.py [hunk 1]
```diff
@@ -191,7 +191,7 @@ def _get_square_data_frame(data, index):
 def write_special(meas_input, phase_advances, plane_tune, plane):
     # TODO REFACTOR AND SIMPLIFY
     accel = meas_input.accelerator
-    meas = phase_advances["MEAS"]
+    meas = phase_advances['free']["MEAS"]
```
Comment (mihofer) [inline omc3/optics_measurements/phase.py:194]: for consistency reasons, wouldn't it be simpler if either 
a) the `["QF"]` selection part in the call would then also be moved into the function, or 
b) the `['free']` part be moved in the fct call?
personally, i'd prefer b) as it keeps the fct more general..

Comment (fsoubelet) [reply omc3/optics_measurements/phase.py:194]: I agree with b

File: omc3/optics_measurements/coupling.py [hunk 2]
```diff
@@ -81,63 +82,67 @@ def calculate_coupling(meas_input, input_files, phase_dict, tune_dict, header_di
...
+    # ----- Old Cminus ----- #
     C_old = 4.0 * tune_sep * np.mean(np.abs(f1001))
     header_dict["OldCminus"] = C_old
     LOG.info(f"abs OldCminus = {C_old}, tune_sep = {tune_sep}")
 
-    # new Cminus
+    # ----- New Cminus ----- #
     C_new = np.abs(4.0 * tune_sep * np.mean(f1001 * exp(1.0j * (joined[f"{COL_MU}X"] - joined[f"{COL_MU}Y"]))))
     header_dict["newCminus"] = C_new
     LOG.info(f"abs NewCminus = {C_new}")
```
[the remark this answers is not in the document]
Comment (fsoubelet) [reply omc3/optics_measurements/coupling.py:118]: I agree, the difference should also be reflected in the issued logs.

[the remark this answers is not in the document]
Comment (awegsche) [reply omc3/optics_measurements/coupling.py:118]: Sorry, wrong pull request and wrong branch.

@fsoubelet sorry for the additionaly work but I kindly request that you apply your style changes another branch (that I will put up in a moment)

[the remark this answers is not in the document]
Comment (awegsche) [reply omc3/optics_measurements/coupling.py:118]: here you go: #322

**Comment by awegsche** [comment]
and here we have the fixed coupling direction. Now we can consolidate changes to `couplilng.py` in here.
**note** tests are yet to come. this is still a draft pull request

**Comment by fsoubelet** [comment]
Let's get on with #324, merge back from master and then get this one merged hopefully within the week.

**Review by fsoubelet — APPROVED** [review]
I'm ok with this PR now

**Review by awegsche — APPROVED** [review]

**Pull request merged**